core/server/worker/src/main/java/alluxio/worker/block/UnderFileSystemBlockReader.java

```diff
@@ -305,18 +303,16 @@ private void updateUnderFileSystemInputStream(long offset) throws IOException {
   }
 
   /**
-   * Updates the block writer given an offset to read. If the offset is beyond the current
-   * position of the block writer, the block writer will be aborted.
-   *
-   * @param offset the read offset
+   * Closes the current block writer, cleans up its temp block and sets it to null.
    */
-  private void updateBlockWriter(long offset) throws IOException {
+  private void cancelBlockWriter() throws IOException {
+    if (mBlockWriter == null) {
+      return;
+    }
     try {
-      if (mBlockWriter != null && offset > mBlockWriter.getPosition()) {
-        mBlockWriter.close();
-        mBlockWriter = null;
-        mLocalBlockStore.abortBlock(mBlockMeta.getSessionId(), mBlockMeta.getBlockId());
-      }
+      mBlockWriter.close();
+      mBlockWriter = null;
+      mLocalBlockStore.abortBlock(mBlockMeta.getSessionId(), mBlockMeta.getBlockId());
```

Comment:
is the exception handling below still appropriate for the new call sites?

Reply:
I think so.